dev-java/protobuf-java/protobuf-java-4.31.0.ebuild

```diff
@@ -0,0 +1,166 @@
+		-e 's:${protobuf.source.dir}:src:' \
+		-e 's:^.*value="::' -e 's:\"/>::' \
+		-e '/project\|echo\|mkdir\|exec/d' \
+		-i java/core/generate-sources-build.xml || die "sed to sources failed"
+
+	# https://github.com/protocolbuffers/protobuf/blob/main/java/core/generate-test-sources-build.xml
+	einfo "Replace variables in generate-test-sources-build.xml"
+	sed \
+		-e 's:${generated.testsources.dir}:java/core/src/test/java:' \
+		-e 's:${protobuf.source.dir}:src:' \
+		-e 's:${test.proto.dir}:java/core/src/test/proto:' \
+		-e 's:^.*value="::' -e 's:\"/>::' \
+		-e '/project\|mkdir\|exec\|Also generate/d' \
+		-i java/core/generate-test-sources-build.xml || die "sed to test sources failed"
+
+	# Split the file in two parts, one for each run-protoc call
+	awk '/--java_out/{x="test-sources-build-"++i;}{print > x;}' \
+		java/core/generate-test-sources-build.xml || die
+
+	# Requires TestParameterInjector library, currently not available in Gentoo.
+	rm java/core/src/test/java/com/google/protobuf/CodedInputStreamTest.java || die
+
+	# java/core/src/test/java/editions_unittest/TestDelimited.java:2867:
+	# error: package editions_unittest.MessageImport does not exist
+	rm java/core/src/test/java/com/google/protobuf/TextFormatTest.java || die
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-Dprotobuf_BUILD_TESTS=OFF
+		-Dprotobuf_LOCAL_DEPENDENCIES_ONLY=ON
+	)
+	if ! use system-protoc; then
+		cmake_src_configure
+	fi
+}
+
+src_compile() {
+	if ! use system-protoc; then
+		cmake_src_compile
+	fi
+
+	einfo "Run protoc to generate sources"
+	run-protoc \
+		@java/core/generate-sources-build.xml \
+		|| die "protoc sources failed"
+
+	java-pkg-simple_src_compile
+}
+
+src_test() {
+	local -x JAVA_GENTOO_CLASSPATH_EXTRA="${DISTDIR}/truth-${TV}.jar"
+
+	# google/protobuf/java_features.proto: File not found.
+	cp {java/core/src/main/resources,src}/google/protobuf/java_features.proto || die
+
+	einfo "Running protoc on first part of generate-test-sources-build.xml"
+	run-protoc @test-sources-build-1 \
+		|| die "run-protoc test-sources-build-1 failed"
+
+	einfo "Running protoc on second part of generate-test-sources-build.xml"
+	run-protoc @test-sources-build-2 \
+		|| die "run-protoc test-sources-build-2 failed"
+
+	einfo "Running tests"
+	# Invalid test class 'map_test.MapInitializationOrderTest':
+	# 1. Test class should have exactly one public constructor
+	# Invalid test class 'protobuf_unittest.CachedFieldSizeTest':
+	# 1. Test class should have exactly one public constructor
+	pushd "${JAVA_TEST_SRC_DIR}" >/dev/null || die
+		local JAVA_TEST_RUN_ONLY=$(find * \
+			-path "**/*Test.java" \
+			! -path "**/Abstract*Test.java" \
+			! -name "MapInitializationOrderTest.java" \
+			! -name CachedFieldSizeTest.java
+			)
+	popd >/dev/null || die
+	JAVA_TEST_RUN_ONLY="${JAVA_TEST_RUN_ONLY//.java}"
+	JAVA_TEST_RUN_ONLY="${JAVA_TEST_RUN_ONLY//\//.}"
+	java-pkg-simple_src_test
+}
+
+src_install() {
+	java-pkg-simple_src_install
+}
```

Comment:
Adding `src_install()` was part of the hack (but `cmake` wrongly at the end of the inherit line) and should also be dropped.
If keeping the hack, sort eclasses alpabetically AND also drop `src_install()`.